config-model/src/main/java/com/yahoo/vespa/model/content/storagecluster/FileStorProducer.java

```diff
@@ -56,10 +58,15 @@ private Integer getThreads(ModelElement clusterElem) {
         }
     }
     public FileStorProducer(ModelContext.Properties properties, ContentCluster parent, Integer numThreads) {
+        final int twoMB = 0x200000;
+        final int fourK = 0x1000;
         this.numThreads = numThreads;
         this.cluster = parent;
         this.reponseNumThreads = properties.defaultNumResponseThreads();
         this.responseSequencerType = convertResponseSequencerType(properties.responseSequencerType());
+        useAsyncMessageHandlingOnSchedule = properties.useAsyncMessageHandlingOnSchedule();
+        mergeChunkSize = ((properties.mergeChunkSize() + (twoMB-1))/twoMB)*twoMB - fourK;
```

Comment:
Also, do we need the -4KiB adjustment with today's allocation strategies? The config comment mentions this as being done to avoid hitting the next malloc size class, but I'm not sure if this is relevant now.

Reply:
Separated out the 2MB alignment and removed the 4k reservation. Also updated the description.